javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AddBaggageSpanProcessor.java

```diff
@@ -0,0 +1,18 @@
+package io.opentelemetry.javaagent.tooling;
+
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+
+public class AddBaggageSpanProcessor implements OnStartSpanProcessor {
+  @Override
+  public void onStart(Context context, ReadWriteSpan span) {
+    Baggage baggage = Baggage.fromContext(context);
+    baggage.forEach(
+        (key, value) ->
+            span.setAttribute(
+                // add prefix to key to not override existing attributes
+                "baggage." + key,
+                value.getValue()));
```

Comment:
Given the lack of spec around this topic, it would be good to keep the attribute naming flexible and have the `baggage` namespace prefix come in through the constructor as an optional parameter. The suggested whitelisting of baggage keys to attach would provide the necessary security to prevent unwanted overrides.